lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -590,3 +653,40 @@ bool SslClientConnection::readable() {
 
   return false;
 }
+
+bool SslClientConnection::setSocketToNonBlocking() {
+#if defined(__linux__) || defined(__APPLE__)
+  if (fcntl(_socket.fileDescriptor, F_SETFL, _socketFlags | O_NONBLOCK) == -1) {
+    _errorDetails = "Attempt to create non-blocking socket generated error " +
+                    std::to_string(errno);
+    return false;
+  }
+#else
+  u_long nonBlocking = 1;
+  if (ioctlsocket(_socket.fileDescriptor, FIONBIO, &nonBlocking) != 0) {
+    _errorDetails = "Attempt to create non-blocking socket generated error " +
+                    std::to_string(WSAGetLastError());
+    return false;
+  }
+#endif
+  return true;
+}
+
+bool SslClientConnection::cleanUpSocketFlags() {
+#if defined(__linux__) || defined(__APPLE__)
```

Comment:
Let's add an assertion here, so that we know it is only called if we are actually using a non-blocking socket.
```suggestion
  TRI_ASSERT(_isSocketNonBlocking);
#if defined(__linux__) || defined(__APPLE__)
```